MonoGame.Framework/Graphics/Texture2D.OpenGL.cs

```diff
@@ -489,8 +490,9 @@ private void PlatformSaveAsJpeg(Stream stream, int width, int height)
 
         private void PlatformSaveAsPng(Stream stream, int width, int height)
         {
-#if MONOMAC || WINDOWS
-            SaveAsImage(stream, width, height, ImageFormat.Png);
+#if MONOMAC || WINDOWS || IOS
+            var pngWriter = new PngWriter();
+            pngWriter.Write(this, stream);
 #elif ANDROID
             SaveAsImage(stream, width, height, Bitmap.CompressFormat.Png);
```

Comment:
Yes, in theory it should work. Don't have access to any Android devices
though.
On Tue, 10 Mar 2015 at 22:09 Tom Spilman <notifications@github.com> wrote:

> In MonoGame.Framework/Graphics/Texture2D.OpenGL.cs
> <https://github.com/mono/MonoGame/pull/3601#discussion_r26160705>:
>
> > @@ -489,8 +490,9 @@ private void PlatformSaveAsJpeg(Stream stream, int width, int height)
> >
> >          private void PlatformSaveAsPng(Stream stream, int width, int height)
> >          {
> > -#if MONOMAC || WINDOWS
> > -            SaveAsImage(stream, width, height, ImageFormat.Png);
> > +#if MONOMAC || WINDOWS || IOS
> > +            var pngWriter = new PngWriter();
> > +            pngWriter.Write(this, stream);
> >  #elif ANDROID
> >              SaveAsImage(stream, width, height, Bitmap.CompressFormat.Png);
>
> What is missing for using this on Android? Testing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mono/MonoGame/pull/3601/files#r26160705>.
>